### DISCUSSION TITLE
How to output SWE (snow water equivelent) and Root Soil Moisture

### DISCUSSION THREAD

**Discussion by ZRYings** [iiasa/CWatM#97] [Q&A] [answered]
Hi,

I am currently conducting research that involves the soil and snow module in CWatM. I am wondering how to extract SWE and Root Soil Moisture from CWatM outputs.

(1) For SWE Variable: I attempted to extract the "SnowWaterEquivalent" variable, but received an error indicating that there were no values for this variable. Instead I used "SnowCover" as SWE but am uncertain about its appropriateness. Additionally, this variable shows an increasing trend over time. Is this reasonable?
(2) For Root Soil Moisture Variable: I was unable to find the variable for Root Soil Moisture. I only found related variables like w1, w2, w3, sum_w1, sum_w2, and sum_w3. How can I obtain Root Soil Moisture data directly from CWatM or through alternate means?

Thanks! Look forward to any reply.

**Comment by mikhailsmilovic** [comment, marked as answer]
**Snow cover**
`SnowCover` is the variable for the depth of moisture [metres] in the snowpack. An infinitely increasing trend over time is not reasonable but has been noticed in areas with glaciers. There have been recent developments to redistribute snow to avoid this. A glacier module in development would resolve this in glacier areas. 

We suggest including this in the settings file in section [SNOW]: 
_snowmelt_radiation = True_
This will induce snowmelt from temperature and radiation.

**Soil moisture**
Soil moisture is as you have: sum_w1, sum_w2, sum_w3 for soil layers 1, 2, and 3, respectively, and  
`self.var.sum_soil = self.var.sum_w1 + self.var.sum_w2 + self.var.sum_w3 + self.var.sum_topwater`

Plants absorb/pull water from these soil layers relative to the fraction of roots in each soil layer, called adjRoot: 
`self.var.adjRoot[soilLayer][i]` = fraction of the roots in soilLayer for land class i (In landcovertype.py)

Snow cover and soil moisture are combined in the following variable:
`self.var.totalSto = self.var.SnowCover + self.var.sum_interceptStor + self.var.sum_soil`
